### PR TITLE
Clean up Kafka Cluster and pubsub topic in rc validation script

### LINKED ISSUE #23020: [Bug]: Kubernetes service not cleaned up and reaching quota causing multiple performance test failure
### What happened?

Multiple performance tests perma-red since Aug 31st, including JDBC, Kafka, IO Tests on HDFS. The cause is the number of existing k8s endpoints has run out of quota. These end points all have namespace beam-performancetests-[testname]-[test-number]. From the test number it can be seen there are all old tests. Usually k8s service gets deleted at the end of the test. But if jenkins worker fails in the middle, which happened on Aug 30, those service is never cleaned up.

Another leak found is that rc validation script creates a k8s cluster named after "xlang-kafka" for kafka taxi test: 
https://github.com/apache/beam/blob/31561e2ff13147aa80f9f811e2a94ebe57b25374/release/src/main/scripts/run_rc_validation.sh#L501
, but never gets deleted. Consequently every time rc validation there is a new cluster created and at some point it will reach cluster quota.

There should be a cron job automatically clean up these services and clusters.

### Issue Priority

Priority: 1

### Issue Component

Component: test-failures

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #23020

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @chamikaramj

**Review by tvalentyn — APPROVED** [review]

**Comment by tvalentyn** [comment]
cc: @lostluck

File: release/src/main/scripts/run_rc_validation.sh [hunk 1]
```diff
@@ -55,6 +55,18 @@ function clean_up(){
 
   rm -rf ${LOCAL_BEAM_DIR}
   echo "* Deleted workspace ${LOCAL_BEAM_DIR}"
+
+  if [[ -n `which gcloud` ]]; then
+    if [[ -n ${KAFKA_CLUSTER_NAME} ]]; then
+        echo "-----------------------Clean up Kafka Cluster on GKE------------------------"
```
Comment (tvalentyn) [inline release/src/main/scripts/run_rc_validation.sh:61]: will this cleanup affect any running tests outside of RC validation?

Comment (Abacn) [reply release/src/main/scripts/run_rc_validation.sh:61]: The variable KAFKA_CLUSTER_NAME is created below in L511. This clean up should only trigger when there is a kafka cluster being created there. Unless the shell already has an environment variable named "KAFKA_CLUSTER_NAME" and used for testing, and do not set `python_xlang_kafka_taxi_dataflow` to true in this test, it is expected not affecting running test.

**Comment by johnjcasey** [comment]
LGTM

**Pull request merged**